### ISSUE TITLE
Change locator to use a interface

### ISSUE THREAD

**Issue by glennawatson** [reactiveui/splat#284]
This will be a breaking change.

Make the locator use a interface.

Eg change it so you access thorugh a member called `Locator.Current`

This will simplify testing dramatially.

I know @rlittlesii wants to wait to v8, thoughts @dpvreony

**Comment by dpvreony** [comment]
I started putting the locator in #282 as 2 pieces to aid with the testing and unblock the current PR's. I am mindful of the size of the change and the potential impact. see if this going in the right direction. i still need to go through the tests on it. Do need to look at the test unit detection piece inside it to see if it's still needed if you're having an internal implementation to test. This piece i'm really mindful on in case it impacts reactiveui testing etc.

I extracted the logic into an internal class for a testable piece
https://github.com/reactiveui/splat/pull/282/commits/e10d5fee4bb999fab324233358f25c7fe86e6f4e#diff-42a9de1e955f735a8941ee000073edec


While leaving the rest of locator as the static public piece that just passes through to the internal class. so other than the setlocator change  the behaviour remains close to the original.
https://github.com/reactiveui/splat/pull/282/commits/e10d5fee4bb999fab324233358f25c7fe86e6f4e#diff-dd10ade5f5e89cc93e29c391e8a00998

I did put an Internal member of Locator.Internal that effectively gives

```
internal IDependencyResolver Internal => _dependencyResolver;
```
